src/System.Net.Requests/tests/HttpWebRequestTest.cs

```diff
@@ -625,8 +617,30 @@ public void HttpWebRequest_Serialize_Fails()
                 BinaryFormatter formatter = new BinaryFormatter();
                 var hwr = HttpWebRequest.CreateHttp("http://localhost");
 
-                Assert.Throws<PlatformNotSupportedException>(() => formatter.Serialize(fs, hwr));
+                if (PlatformDetection.IsFullFramework)
+                {
+                    // .NET Framework throws a more detailed exception.
```

Comment:
Binary Serialization is now supported on .NET Core 2.0 but it wasn't in 1.1 or 1.0; so probably that is why we were throwing PNSE, so I think the same thing, it sounds reasonable to throw SE.